File: Backend/Agronexis.Model/EntityModel/Product.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Agronexis.Model.EntityModel
{
    [Table("Product", Schema = "dbo")]
    public class Product
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public Guid Id { get; set; }
        public string? Name { get; set; }
        public string? Description { get; set; }
        public string? ImageUrls { get; set; }
        public string? KeyFeatures { get; set; }
        public string? Uses { get; set; }
        public string? MetaTitle { get; set; }
        public string? MetaDescription { get; set; }
        public string? Origin { get; set; }
        public string? ShelfLife { get; set; }
        public string? StorageInstructions { get; set; }
        public string? Certifications { get; set; }
        public string? Ingredients { get; set; }
        public string? NutritionalInfo { get; set; }
        public Guid BrandId { get; set; }
        [ForeignKey("Category")]
        public Guid CategoryId { get; set; }
        public DateTime? ManufacturingDate { get; set; }
        public DateTime? CreatedDate { get; set; }
        public DateTime? ModifiedDate { get; set; }
        public bool IsPremium { get; set; }
        public bool IsFeatured { get; set; }
        public bool IsActive { get; set; }
        public bool IsDeleted { get; set; }
        public Category Category { get; set; }
        public ICollection<Inventory> Inventories { get; set; }

    }
}

File: UI/src/types/product.ts
export interface PriceAndSku {
  id: string;
  price: number;
  isDiscounted: boolean;
  discountPercentage: number;
  discountedAmount: number;
  skuNumber: string;
  weightId: string;
  weightValue: number;
  weightUnit: string;
  currencyCode: string;
  currencyId: string;
  barcode: string;
  createdDate: string;
  modifiedDate: string | null;
  isActive: boolean;
  isDeleted: boolean;
}

export interface Product {
  id: string;
  name: string;
  description: string;
  brandId: string;
  categoryId: string;
  categoryName: string;
  manufacturingDate: string;
  hsnCode: string;
  createdDate: string;
  modifiedDate: string | null;
  isActive: boolean;
  isDeleted: boolean;
  metaTitle: string;
  metaDescription: string;
  imageUrls: string[];
  keyFeatures: string[];
  uses: string[];
  origin: string;
  shelfLife: string;
  storageInstructions: string;
  certifications: string[];
  isPremium: boolean;
  isFeatured: boolean;
  ingredients: string;
  nutritionalInfo: string;
  thumbnailUrl?: string;
  pricesAndSkus: PriceAndSku[];
}
export interface ProductFormData {
  // Product fields
  id?: string;
  name: string;
  description: string;
  brandId?: string;
  brand?: string;
  categoryId?: string;
  category?: string;
  categoryName?: string;
  manufacturingDate?: string;
  hsnCode?: string;
  createdDate?: string;
  modifiedDate?: string | null;
  isActive?: boolean;
  isDeleted?: boolean;
  metaTitle?: string;
  metaDescription?: string;
  imageUrls?: (string | File)[];
  keyFeatures: string[];
  uses: string[];
  origin: string;
  shelfLife: string;
  storageInstructions: string;
  certifications: string[];
  isPremium: boolean;
  isFeatured: boolean;
  ingredients?: string;
  nutritionalInfo?: string;
  thumbnailUrl?: string;
  thumbnailImage?: File | null;
  stock?: number;
  // PriceAndSku fields (as array)
  pricesAndSkus: Array<{
    id?: string;
    price: number;
    isDiscounted: boolean;
    discountPercentage: number;
    discountedAmount: number;
    skuNumber: string;
    weightId: string;
    weightValue: number;
    weightUnit: string;
    currencyCode: string;
    currencyId: string;
    barcode: string;
    createdDate?: string;
    modifiedDate?: string | null;
    isActive?: boolean;
    isDeleted?: boolean;
  }>;
}

export interface BrandInfo {
  isSuccess: boolean;
  code: string;
  message: string;
}

export interface BrandData {
  id: string;
  name: string;
  code: string;
  description: string | null;
  logoUrl: string | null;
  createdDate: string;
  modifiedDate: string | null;
  isActive: boolean;
  isDeleted: boolean;
  metaTitle: string | null;
  metaDescription: string | null;
}

export interface Brand {
  info: BrandInfo;
  id: string | null;
  data: BrandData[];
}

export interface FormattedBrand {
  label: string;
  value: string;
}

export interface Category {
  id: string;
  name: string;
  imageUrl: string;
  isActive: boolean;
}

export interface CategoryFormData {
  name: string;
  description: string;
  brandId?: string;
  metaTitle?: string;
  metaDescription?: string;
  isActive?: boolean;
  imageUrl?: string;
}

export interface BrandFormData {
  name: string;
  code?: string;
  description?: string;
  metaTitle?: string;
  metaDescription?: string;
  isActive?: boolean;
  logoUrl?: string;
}

export interface FormattedCategory {
  id: string;
  title: string;
  image: string;
}

export interface Weight {
  id: string;
  value: number;
  unit: string;
}

export interface FormattedWeight {
  label: string;
  value: string;
}

export interface Currency {
  id: string;
  code: string;
  name: string;
}

export interface FormattedCurrency {
  label: string;
  value: string;
}
